sklearn/preprocessing/_encoders.py

```diff
@@ -179,6 +227,9 @@ class OneHotEncoder(_BaseEncoder):
         Categories (unique values) per feature:
 
         - 'auto' : Determine categories automatically from the training data.
+        - 'dtypes' : Uses pandas categorical dtype to encode categories. For
+          non pandas categorical data, the categories are automatically
+          determined from the training data.
```

Comment:
maybe this could be just part of 'auto' as mentioned in the comments.

But if we keep the 'dtypes' option, we should probably raise an error if a non-df is passed in.